examples/polyscope-examples/exampleHarmonicParametrization.cpp

```diff
@@ -0,0 +1,353 @@
+ * @author Baptiste GENEST (\c baptistegenest@gmail.com )
+ * internship at Laboratoire d'InfoRmatique en Image et Systemes d'information - LIRIS (CNRS, UMR 5205), CNRS, France
+ *
+ * @date 2022/06/15
+ *
+ * This file is part of the DGtal library.
+ */
+
+//////// INCLUDES ////////
+
+#include <iostream>
+
+#include <DGtal/base/Common.h>
+#include <DGtal/helpers/StdDefs.h>
+#include <DGtal/helpers/Shortcuts.h>
+#include <DGtal/helpers/ShortcutsGeometry.h>
+#include <DGtal/shapes/SurfaceMesh.h>
+#include <DGtal/geometry/surfaces/DigitalSurfaceRegularization.h>
+#include <DGtal/dec/PolygonalCalculus.h>
+#include <DGtal/dec/VectorsInHeat.h>
+
+#include <polyscope/polyscope.h>
+#include <polyscope/surface_mesh.h>
+#include <polyscope/point_cloud.h>
+#include <polyscope/curve_network.h>
+
+#include <Eigen/Dense>
+#include <Eigen/Sparse>
+
+////////// NAMESPACES /////////
+
+using namespace DGtal;
+
+////////// TYPEDEFS /////////
+// Using standard 3D digital space.
+typedef Shortcuts<Z3i::KSpace>         SH3;
+typedef ShortcutsGeometry<Z3i::KSpace> SHG3;
+// The following typedefs are useful
+typedef SurfaceMesh< Z3i::RealPoint, Z3i::RealVector >  SurfMesh;
+typedef SurfMesh::Vertices                    Vertices;
+typedef SurfMesh::RealPoint                   RealPoint;
+typedef SurfMesh::Face                        Face;
+typedef SurfMesh::Vertex                      Vertex;
+typedef PolygonalCalculus<SH3::RealPoint,SH3::RealVector> PC;
+typedef PC::SparseMatrix SparseMatrix;
+typedef PC::DenseMatrix DenseMatrix;
+typedef PC::Solver Solver;
+typedef PC::Vector Vector;
+typedef Eigen::Triplet<double> Triplet;
+typedef std::vector<Vertex> chain;
+
+
+//Polyscope global
+polyscope::SurfaceMesh *psMesh;
+polyscope::SurfaceMesh *psParam;
+polyscope::PointCloud *psVertices;
+polyscope::CurveNetwork *psBoundary;
+
+//DEC
+PC *calculus;
+VectorsInHeat<PC> *VHM;
+
+//surface mesh global
+SurfMesh surfmesh;
+std::vector<std::vector<size_t>> faces;
+std::vector<RealPoint> positions;
+
+/**
+ * @brief block
+ * @return the sub sparse matrix that starts at (row,col) of size (height,width)
+ */
+SparseMatrix block(const SparseMatrix& mat, size_t row, size_t col,size_t height,size_t width)
+{
+    SparseMatrix B(height,width);
+    auto last_col = col + width;
+    auto last_row = row + height;
+    std::vector<Triplet> T;
+    for (size_t i = col;i<last_col;i++)
+        for (SparseMatrix::InnerIterator it(mat, i); it; ++it)
+            if (it.row() >= (long)row){
+                if (it.row() < (long)last_row){
+                    T.push_back(Triplet(it.row()-row,i-col,it.value()));
+                }
+                else
+                    break;
+            }
+    B.setFromTriplets(T.begin(),T.end());
+    return B;
+}
+
+/**
+ * @brief computeManifoldBoundaryChains
+ * @return returns the list of chains that borders the mesh
+ */
+std::vector<chain> computeManifoldBoundaryChains(int nb_chains = -1){
```

Comment:
in the coding style,   `{` are on a new line.
Cf https://github.com/DGtal-team/DGtal/blob/master/CONTRIBUTING.md